freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaHealthDetailsService.java

```diff
@@ -53,35 +57,69 @@
 
     public HealthDetailsFreeIpaResponse getHealthDetails(String environmentCrn, String accountId) {
         Stack stack = stackService.getByEnvironmentCrnAndAccountIdWithLists(environmentCrn, accountId);
-        InstanceMetaData master = findMaster(stack);
-        Optional<RPCResponse<Boolean>> rpcResponse = Optional.empty();
-        try {
-            rpcResponse = Optional.ofNullable(checkFreeIpaHealth(stack, master));
-        } catch (FreeIpaClientException e) {
-            LOGGER.error("Unable to check the health of FreeIPA.", e);
+        List<InstanceMetaData> instances = stack.getInstanceGroups().stream()
+                .flatMap(instanceGroup -> instanceGroup.getInstanceMetaData().stream()).collect(Collectors.toList());
+        HealthDetailsFreeIpaResponse response = new HealthDetailsFreeIpaResponse();
+        // The code here is weird because the server_conncheck API requires a specific node
+        // and we don't know what determines what that node needs to be.  But if we use
+        // the wrong master, we get an exception that says
+        //   invalid 'cn': must be "ipaserver0.xx.wl.cloudera.site"
+        // so we parse that, set that as master, and then retry the node.
```

Comment:
why don't we know which instance we should use?

Reply:
It's not stored in the stack data, and I haven't been able to determine which one it should be.  If we have multi-master setup, it's not clear which one is the real master.

Reply:
I think this is something we'll need to add in the next round of HA work.